hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/HoodieSparkUtils.scala

```diff
@@ -324,7 +326,14 @@ object HoodieSparkUtils extends SparkAdapterSupport {
       val name2Fields = tableAvroSchema.getFields.asScala.map(f => f.name() -> f).toMap
       // Here have to create a new Schema.Field object
       // to prevent throwing exceptions like "org.apache.avro.AvroRuntimeException: Field already used".
-      val requiredFields = requiredColumns.map(c => name2Fields(c))
+      val requiredFields = requiredColumns.filter(c => {
```

Comment:
We should not relax this here actually, b/c `requiredColumns` will contain also query columns.

Instead we should provide `HoodieMergeOnReadRDD` 2 parquet readers: 

1. Primed for merging (ie for schema containing record-key, precombine-key)
2. Primed for NO merging (ie whose schema could be essentially empty)

Reply:
can't we do this while appending mandatory columns ? i.e compare w/ table schema and drop missing fields. so that we do this filtering only for the mandatory columns that we look to add and not touch query columns.

Reply:
alexey: I am not very sure on the amount of changes required for the proposal you have made. but lets try to make minimal changes to make progress w/o requiring more testing. anyways, we will revisit the preCombine field setting altogether for 0.12 and put some fixes.

Reply:
I'm going to rethink the minimal changes to unblock 0.11 release.  The changes in the current shape introduce the problem with non-existent query columns as you mentioned.